### ISSUE TITLE
Windows IPv6 Support

### ISSUE THREAD

**Issue by theonewolf** [ronf/asyncssh#49]
@ronf default socket binding on Windows dies on me with the following error (using `devel` branch, Windows 8.1 64-bit, Python 3.4 64-bit):

```bash
$ python forwarder.py
Traceback (most recent call last):
  File "forwarder.py", line 17, in <module>
    get_event_loop().run_until_complete(run_client())
  File "C:\Python34\lib\asyncio\base_events.py", line 316, in run_until_complete
    return future.result()
  File "C:\Python34\lib\asyncio\futures.py", line 275, in result
    raise self._exception
  File "C:\Python34\lib\asyncio\tasks.py", line 236, in _step
    result = coro.send(value)
  File "forwarder.py", line 11, in run_client
    listener = yield from conn.forward_local_port('', 8450, '127.0.0.1', 8450)
  File "C:\Python34\lib\site-packages\asyncssh\connection.py", line 1874, in forward_local_port
    listen_port))
  File "C:\Python34\lib\site-packages\asyncssh\listener.py", line 217, in create_tcp_forward_listener
    sock.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, True)
AttributeError: 'module' object has no attribute 'IPPROTO_IPV6'
```

I was trying a simple TCP port forwarding:

```python
#!/usr/bin/env python3
from asyncio import coroutine, get_event_loop
from getpass import getpass
from sys import exit

from asyncssh import connect, Error

@coroutine
def run_client():
    with (yield from connect('something.com', username='ubuntu')) as conn:
        listener = yield from conn.forward_local_port('', 8080, '127.0.0.1', 8080)
        yield from listener.wait_closed()

    yield from conn.wait_closed()

try:
    get_event_loop().run_until_complete(run_client())
except (OSError, Error) as exc:
    exit('SSH connection failed: ' + str(exc))
```

Workaround is to pass a non-empty, IPv4-based IP address as the first argument to `conn.forward_local_port`.

This is apparently an issue in Python itself:

http://bugs.python.org/issue6926

Where they suggest you hardcode the constant, or otherwise avoid its use.

Old Tornado discussion:

https://groups.google.com/forum/#!topic/python-tornado/fBsiswcLL9E

**Comment by ronf** [comment]
Ok - I've put a guard around the call to set IPV6_V6ONLY. Supposedly, this call should not be needed on Windows as that's already the default behavior. Commit d28da2624b9645617c955dec17a76c8c93c8bbf7 in the develop branch should have this fix.

**Comment by theonewolf** [comment]
Default behavior on modern Windows?

That's all I really care about I guess.

Anyways this issue isn't really urgent as the workaround is pretty trivial.

I'll check the new commit tomorrow morning (US EST).

**Comment by ronf** [comment]
Thanks.

You're right that you could work around the issue if you provide a IPv4-only listening address (or a hostname that resolves to only IPv4 addresses). However, listening on all addresses or on an address or hostname that resolves to IPv6 would trigger the problem (before this change).

**Comment by ronf** [comment]
The fix for this and other recent fixes to run on Windows should now be released as part of version 1.5.1.
